### ISSUE TITLE
Appointments created from 'shared' accounts fail with invalid email uri

### ISSUE THREAD

**Issue by edmcdonagh** [aluxnimm/outlookcaldavsynchronizer#244]
Possibly related to #28?
```
Invalid email address Uri '/O=My institution/OU=EXCHANGE ADMINISTRATIVE GROUP (ABCDEF1234)/CN=RECIPIENTS/CN=Calendar Name QB80' for organizer
```
Followed by:

```<ExceptionThatLeadToAbortion>CalDavSynchronizer.DataAccess.WebDavClientException: Response status code does not indicate success: '500' ('Internal Server Error'). Message:
&lt;?xml version="1.0" encoding="utf-8"?&gt;
&lt;d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns"&gt;
  &lt;s:exception&gt;Sabre\VObject\ITip\SameOrganizerForAllComponentsException&lt;/s:exception&gt;
  &lt;s:message&gt;Every instance of the event must have the same organizer.&lt;/s:message&gt;
&lt;/d:error&gt;

   at CalDavSynchronizer.DataAccess.HttpClientBasedClient.WebDavClient.&lt;EnsureSuccessStatusCode&gt;d__10.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at CalDavSynchronizer.DataAccess.HttpClientBasedClient.WebDavClient.&lt;ExecuteWebDavRequest&gt;d__9.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at CalDavSynchronizer.DataAccess.HttpClientBasedClient.WebDavClient.&lt;ExecuteWebDavRequestAndReturnResponseHeaders&gt;d__8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at CalDavSynchronizer.DataAccess.CalDavDataAccess.&lt;CreateNewEntity&gt;d__18.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at CalDavSynchronizer.Implementation.CalDavRepository`1.&lt;Create&gt;d__17.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GenSync.EntityRepositories.BatchEntityRepositoryAdapter`4.&lt;PerformOperations&gt;d__3.MoveNext()</ExceptionThatLeadToAbortion>
      <Operation>CreateInB</Operation>
```

Appointments that originate in the shared calendar do not synchronise. When I attempted to synchronise that calendar itself, nothing synchronised.

Thanks for your help.

**Comment by aluxnimm** [comment]
How is this meeting with this organizer '/O=My institution/OU=EXCHANGE ADMINISTRATIVE GROUP (ABCDEF1234)/CN=RECIPIENTS/CN=Calendar Name QB80' created?

Is the organizer a special contact in an exchange adressbook or GAL or distribution list?

**Comment by edmcdonagh** [comment]
Hi @aluxnimm
The meeting was created in a shared calendar by a colleague with a standard account. However, it was created a few years ago, recurring indefinitely once a month, and many of the individual meetings in the series had been moved by any one of our group.

After creating this issue, I established that actually it was only this appointment series that was an issue - all the other meetings booked in the shared calendar were syncing fine from my calendar.

I have therefore put an end to the appointment series and replaced it with individual appointments from the beginning of 2018 onwards and now everything syncs without error.

So for me, the issue can be closed. I presume the trigger for the error might be that individual appointments in an appointment series have been changed by different people, but I don't know. I don't know if the fact the appointment was in a shared calendar had anything to do with it.

I don't know if this can be handled better - I am happy for you to close the issue if you see fit.

Thank you and the contributers for such a great tool. Much appreciated 😁
